proxy/grpcproxy/lease.go

```diff
@@ -15,20 +15,71 @@
 package grpcproxy
 
 import (
+	"io"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"google.golang.org/grpc/metadata"
+
 	"golang.org/x/net/context"
+	"golang.org/x/time/rate"
 
 	"github.com/coreos/etcd/clientv3"
+	"github.com/coreos/etcd/etcdserver/api/v3rpc/rpctypes"
 	pb "github.com/coreos/etcd/etcdserver/etcdserverpb"
 )
 
 type leaseProxy struct {
-	client *clientv3.Client
+	client   *clientv3.Client
```

Comment:
LeaseGrant(), LeaseRevoke(),  and LeaseTimeToLive() use client to create new conn.
e.g
```go
func (lp *leaseProxy) LeaseGrant(ctx context.Context, cr *pb.LeaseGrantRequest) (*pb.LeaseGrantResponse, error) {
	conn := lp.client.ActiveConnection()
	return pb.NewLeaseClient(conn).LeaseGrant(ctx, cr)
}
```